src/Type/Definition/Type.php

```diff
@@ -314,10 +315,25 @@ public static function isAbstractType($type) : bool
      */
     public static function assertType($type)
     {
-        Utils::invariant(
-            self::isType($type),
-            'Expected ' . Utils::printSafe($type) . ' to be a GraphQL type.'
-        );
+        $isType = self::isType($type);
+        if (! $isType) {
+            Utils::invariant(
+                $isType,
+                'Expected ' . Utils::printSafe($type) . ' to be a GraphQL type.'
+            );
+        }
+
+        return $type;
+    }
+
+    /**
+     * @param callable|Type $type
+     */
+    public static function resolveLazyType($type)
+    {
+        if (is_callable($type)) {
+            return $type();
+        }
```

Comment:
I guess we need to throw if loaded value is not a type similar to how we had it in the past:

https://github.com/webonyx/graphql-php/blob/c11f25794a16863ad727c844ca9083bf1cb3c3d1/src/Type/Definition/Type.php#L202-L207

Reply:
Ah, no, actually I just type-hinted the return as `Type`, so we can let natural language-level checking take care of this for us.